pkg/combustion/templates/31-elemental-register.sh.tpl

```diff
@@ -47,5 +47,113 @@ Environment="CATTLE_AGENT_CONFIG=/etc/rancher/elemental/agent/config.yaml"
 ExecStart=/usr/sbin/elemental-system-agent sentinel
 EOF
 
+cat <<- EOF > /etc/systemd/system/elemental-reset.path
+[Path]
+PathModified=/var/lib/elemental/.unmanaged_reset
+
+[Install]
+WantedBy=multi-user.target
+EOF
+
+cat <<- EOF > /etc/systemd/system/elemental-reset.service
+[Unit]
+Description=Elemental Reset for Unmanaged Hosts
+Wants=network-online.target
+After=network-online.target
+
+[Service]
+Type=simple
+ExecStart=/opt/edge/node_cleanup.sh -u
+ExecStartPost=/usr/bin/rm -f /var/lib/elemental/.unmanaged_reset
+EOF
+
+systemctl enable elemental-reset.path || true
 systemctl enable elemental-register-systemd.service || true
 systemctl enable elemental-system-agent.service || true
+
+mkdir -p /opt/edge/
+cat <<- \EOF > /opt/edge/node_cleanup.sh
```

Comment:
Embedding the script like this doesn't sit right with me but I'm fine with merging this and seeing if future modifications complicate it enough to warrant extracting it out.

Reply:
Agreed and I didn't see your comment before making mine

Reply:
Yeah, I agree, we should revisit this later and package it properly. Reason being, it gives us a way of answering the two questions at the start. But for now, this gives us a way forward.